src/chess/board.h

```diff
@@ -136,6 +136,23 @@ class ChessBoard {
 
     void Mirror() { data_ = ((data_ & 0b11) << 2) + ((data_ & 0b1100) >> 2); }
 
+    std::string as_string() const {
+      if (data_ == 0) return "-";
+      std::string result;
+      if (queenside_rook() == FILE_A && kingside_rook() == FILE_H) {
```

Comment:
Seems to work fine in a few tests (mostly lichess and cutechess). It can be changed, but is it worth it? E.g. start position 508 would be `rbqnknbr/pppppppp/8/8/8/8/PPPPPPPP/RBQNKNBR w KQkq - 0 1` if we also check the king's position rather than `rbqnknbr/pppppppp/8/8/8/8/PPPPPPPP/RBQNKNBR w AHah - 0 1` and getting this right is not trivial. (Also putting the later form in the lichess board editor loses black's castling rights).
So I'd rather leave it like that until we get actual issues reported.